From d0c4312c3d0517e94ed74c068afdfa5027951fab Mon Sep 17 00:00:00 2001
From: Michael Byrne <michaelx.byrne@intel.com>
Date: Fri, 30 Jun 2017 11:05:44 +0200
Subject: [PATCH 0466/1060] drm/i915/gvt: Fix to remove warning from
 verify_connector_state on DomU start up.

Fixes RTC 19537: 'verify_connector_state throw errors in DomU'

Added a check in intel_ddi_connector_get_hw_state for intel_vgpu_active() when
port A is DP. This corresponds to check added in commit
3d3603f17bed830641f5eec00dc54d7955f9a74b.
Notice: this is to apply to the guest.

Change-Id: Ib7b3e911d4ca9b6e11a6ec1e10be8807550fc030
Reviewed-by: Singh, Satyeshwar <satyeshwar.singh@intel.com>
Reviewed-by: He, Min <min.he@intel.com>
Reviewed-by: Jiang, Fei <fei.jiang@intel.com>
Reviewed-by: Dong, Eddie <eddie.dong@intel.com>
Tested-by: Dong, Eddie <eddie.dong@intel.com>
---
 drivers/gpu/drm/i915/intel_ddi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_ddi.c b/drivers/gpu/drm/i915/intel_ddi.c
index 17feb1112a77..3d8baee642c1 100644
--- a/drivers/gpu/drm/i915/intel_ddi.c
+++ b/drivers/gpu/drm/i915/intel_ddi.c
@@ -1554,7 +1554,7 @@ bool intel_ddi_connector_get_hw_state(struct intel_connector *intel_connector)
 		goto out;
 	}
 
-	if (port == PORT_A)
+	if (port == PORT_A && !intel_vgpu_active(dev_priv))
 		cpu_transcoder = TRANSCODER_EDP;
 	else
 		cpu_transcoder = (enum transcoder) pipe;
-- 
2.17.1

